plugins/plugin-installer/plugin_installer/tasks.py

```diff
@@ -110,57 +126,85 @@ def extract_module_paths(url):
+        return plugin_dir
+    except Exception as e:
+        if os.path.exists(plugin_dir):
+            shutil.rmtree(plugin_dir)
+        raise NonRecoverableError('Failed to download and unpack plugin from '
+                                  '{0}. Reported error: {1}'
+                                  .format(plugin_url, e.message))
+
+
+def extract_plugin_name(plugin_dir):
     previous_cwd = os.getcwd()
-    fetch_plugin_from_pip_by_url = not os.path.isdir(plugin_url)
-    plugin_dir = plugin_url
+
     try:
-        if fetch_plugin_from_pip_by_url:
-            plugin_dir = tempfile.mkdtemp()
-            req_set = pip.req.RequirementSet(build_dir=None,
-                                             src_dir=None,
-                                             download_dir=None)
-            req_set.unpack_url(link=pip.index.Link(plugin_url),
-                               location=plugin_dir,
-                               download_dir=None,
-                               only_download=False)
-        runner = LocalCommandRunner(host=utils.get_local_ip())
         os.chdir(plugin_dir)
+        runner = LocalCommandRunner(host=utils.get_local_ip())
         plugin_name = runner.run(
             '{0} {1} {2}'.format(_python(),
                                  path.join(
                                      path.dirname(__file__),
                                      'extract_package_name.py'),
                                  plugin_dir)).std_out
-        runner.run('{0} install --no-deps {1}'.format(_pip(), plugin_dir))
         return plugin_name
     finally:
         os.chdir(previous_cwd)
-        if fetch_plugin_from_pip_by_url:
-            shutil.rmtree(plugin_dir)
 
 
-def get_url(blueprint_id, plugin):
+def get_url_and_args(blueprint_id, plugin_dict):
 
-    source = plugin['source']
+    source = plugin_dict['source']
+    if source:
+        source = str(source).strip()
+    else:
+        raise NonRecoverableError('Plugin source is missing')
+
+    install_args = ''
+    if 'installation_args' in plugin_dict:
+        install_args = plugin_dict['installation_args']
+
+    if install_args:
+        install_args = str(install_args).strip()
+
+    # validate source url
     if '://' in source:
         split = source.split('://')
         schema = split[0]
-        if schema in ['http', 'https']:
-            # in case of a URL, return as is.
-            return source
-        # invalid schema
-        raise NonRecoverableError('Invalid schema: {0}'.format(schema))
-
-    # Else, assume its a relative path from <blueprint_home>/plugins
-    # to a directory containing the plugin project.
-    # in this case, the archived plugin will reside in the manager file server.
-
-    blueprint_plugins_url = '{0}/{1}/plugins'.format(
-        utils.get_manager_file_server_blueprints_root_url(),
-        blueprint_id
-    )
-    return '{0}/{1}.zip'.format(blueprint_plugins_url, source)
+        if not (schema in ['http', 'https']):
```

Comment:
this kind of condition is usually written as

if schema not in ['http', 'https']